tests/ignite/metrics/test_precision.py

```diff
@@ -47,168 +47,133 @@ def test_binary_wrong_inputs():
         pr.update((torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10, 5, 6)).long()))
 
 
-def test_binary_input_N():
+@pytest.mark.parametrize("average", [False, True])
+def test_binary_input_N(average):
```

Comment:
Another comment, maybe we can refactor all binary `_test` methods ?

Reply:
I think this is a good idea!